YololCompetition/Services/Challenge/IChallenges.cs

```diff
@@ -25,6 +25,8 @@ public interface IChallenges
         IAsyncEnumerable<Challenge> GetChallenges(ChallengeDifficulty? difficultyFilter = null, ulong? id = null, string? name = null, bool includeUnstarted = false);
 
         Task<int> Delete(ulong challengeId);
+        
+        Task<Challenge?> GetChallengesByEndTime(ulong EndUnixTime);
```

Comment:
parameters should start with a lower case letter: `endUnixTime`

Reply:
Also, since this only gets a single challenge, it should be singular `GetChallengeByEndTime`

Reply:
Actually looking more closely at the implementation, the name is misleading. See comments on impl.

Reply:
Fixed. Intentionally gets more than 1 to possibly have more uses for the command (get all challenges run in the last year, etc.)